napari/utils/misc.py

```diff
@@ -131,10 +131,19 @@ def ensure_iterable(
     arg: Union[None, str, Enum, float, list, npt.NDArray], color: bool = False
 ):
     """Ensure an argument is an iterable. Useful when an input argument
-    can either be a single value or a list. If a color is passed then it
-    will be treated specially to determine if it is iterable.
+    can either be a single value or a list.
+    Argument color is deprecated.
     """
-    if is_iterable(arg, color=color):
+    # deprecate color
+    if color:
+        warnings.warn(
+            trans._(
+                'Argument color is deprecated.',
+            ),
+            category=DeprecationWarning,
+            stacklevel=2,  # not sure what level to use here
+        )
+    if is_iterable(arg):
```

Comment:
You likely want to do something like:


```
sentinel = object()

def ensure_iterabel(...., color = sentinel):
   if color is not sentinel:
       warning.warn...
```

This way the warning is triggered even if `color=False`  is passed. 

another way is to do

```
def ensure_iterable(..., *args, **kwargs):
```
and then check the content of args and kwags, but it's a bit more verbose.

You likely also want to say either since when it's deprecated, and/or when it will be removed I think. 

I'm not that involved in the project anymore but other dev will likely pitch in. Check with them.

Reply:
Thanks for the draft PR @palec87 

Agree, though I think we would need to make the decision on when it would be deprecated. Given that it is not used this could perhaps be sooner rather than later (0.5.0). What do you think @Czaki?

Reply:
Deprecation warnings have to be released at least once so people can see them, so we can't deprecate in 0.5.0!

Reply:
> Deprecation warnings have to be released at least once so people can see them, so we can't deprecate in 0.5.0!

Are yon confusing Deprecation and Removal ? 

I mean most of the "Deprecated in X" are emitted for the first when doing release X...

Reply:
Targeted for deprecation from 0.5.0, but then deprecated before 0.6.0 given that these are not used.

Reply:
Sorry, indeed getting confused, ignore my brainfart :P Yes, deprecate in 0.5.0 is good!

Reply:
Thank you @Carreau, I have already learned something. @melonora, with my new commit, if I understand the deprecation logic, I returned the original color argument functionality and only marked it for removal in 0.6.0. It does not change the runtime, because it is not used, however it should be removed only after 0.5.0 release, right?

Reply:
Thanks @palec87, it seems good to me. The stacklevel determines how many calls to unwind so it is good not to get this number too high.